meerk40t/core/elements/elements.py

```diff
@@ -1345,6 +1360,110 @@ def create_image(oplist):
         if needs_signal:
             self.signal("default_operations")
 
+    def create_usable_copy(self, sourceop):
+        from copy import copy
+
+        op_to_use = copy(sourceop)
+        for attr in ("id", "label", "color", "lock", "allowed_attributes"):
+            setattr(op_to_use, attr, getattr(sourceop, attr))
+        return op_to_use
+
+    def set_default_id(self, targetop):
+        id_candidate = 0
+        pattern = "D"  # Default
+        op_name_parts = targetop.type.split(" ")
+        if len(op_name_parts) > 1:
+            pattern = op_name_parts[1].upper()[0]
+        id_existing = True
+        while id_existing:
+            id_existing = False
+            id_candidate += 1
+            op_id = f"{pattern}{id_candidate}"
+            for op in list(self.ops()):
+                if op_id == op.id:
+                    id_existing = True
+                    break
+        targetop.id = op_id
+
+    def assign_default_operation(self, data, targetop):
+        emphasize_mode = False
+        if data is None:
+            emphasize_mode = True
+            data = list(self.flat(emphasized=True))
+        if len(data) == 0:
+            return
+        emph_data = [e for e in data]
+        op_id = targetop.id
+        if op_id is None:
+            # WTF, that should not be the case
```

Comment:
If a thing shouldn't happen. Crashing may be correct, then fixing the root cause. I'm guessing since the default ops were being moved directly into tree, the modifications of the tree could clear out the ID and cause issues. If I'm reading the PR correctly.

Reply:
As I have changed a couple of things to address the root causes (wrong undo, direct attachment of a default operation to the tree instead of creating and using a copy), that should really never happen. But it does not hurt either...